src/quo2/components/token_tag.cljs

```diff
@@ -0,0 +1,68 @@
+(ns quo2.components.token-tag
+  (:require [quo2.foundations.colors :as colors]
+            [quo.react-native :as rn]
+            [quo.theme :as theme]
+            [status-im.ui.components.icons.icons :as icons]
+            [quo2.components.text :as text]))
+
+(def themes {:light   {:background-color colors/neutral-20}
+             :dark  {:background-color colors/neutral-80}})
+
+(defn get-value-from-size [size, big-option, small-option]
+  (if  (= size :big) big-option small-option))
+
+(defn style-container [size border-color is-required]
+  (merge {:height             32
+          :min-width         (get-value-from-size size 94 76)
+          :align-items        :center
+          :flex-direction     :row
+          :left             0
+          :border-radius      100
+          :padding-right      10}
+         (when is-required {:border-color     border-color
+                          :border-width     1} )))
+
+(defn token-tag
+  "[token-tag opts \"label\"]
+   opts
+   {
+    :token string
+    :value string
+    :size :small/:big
+    :icon-name :icon-name
+    :border-color :color
+    :is-required true/false
+    :is-purchasable true/false
+    }"
+  [_ _]
+  (fn [{:keys [token value size icon-name border-color is-required is-purchasable] :or {size :small}}]
+    [rn/view {:style   (merge (style-container size border-color is-required)  (get-in themes [(theme/get-theme)]))}
```

Comment:
No need to use `(get-in themes [(theme/get-theme)])`, if not a nested map, you can use `get`

`(get themes (theme/get-theme))`

or (fun facts)

`(themes (theme/get-theme))`
`((theme/get-theme) themes))`

as both `keywords` and `maps` implement `IFn` (https://clojure.github.io/clojure/javadoc/clojure/lang/IFn.html ) and implement `get`.

`(get themes (theme/get-theme))` is the most readable :)

Reply:
awesome, thanks Andrea! :)

Reply:
updated